src/cuepoints.ts

```diff
@@ -0,0 +1,38 @@
+import {CuepointService} from './cuepoint-service';
+import Player = KalturaPlayerTypes.Player;
+
+export class Cuepoints extends KalturaPlayer.core.BasePlugin {
+  private _cuePointService: CuepointService;
+  /**
+   * The default configuration of the plugin.
+   * @static
+   */
+  static defaultConfig = {};
+
+  constructor(name: string, player: Player) {
+    super(name, player);
+    this.eventManager.listen(this.player, this.player.Event.CHANGE_SOURCE_ENDED, () => {
+      this._cuePointService.init();
```

Comment:
we init the service on every change media? this is a service so why init it each time?
Maybe semantics are off but would expect to have it reset on each change media, which we already do below so why init?
Update: OK, I see now what you did, so you need to know if media type is live or vod.
I think this is an internal service thing so make more sense to register events of internal state that are not covered by plugin life cycle(init, reset, destroy) inside service.

Reply:
done